core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -219,6 +220,19 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
+  override def getEventLogPaths(appId: String, attemptId: Option[String]): Seq[Path] = {
+
+    val filePaths = new ArrayBuffer[Path]()
+    applications.get(appId).foreach { appInfo =>
```

Comment:
Since `applications.get(appId)` returns an `Option`, the `flatMap` won't work. We'd need to do a `map` with the same code inside followed by a `getOrElse(Seq.empty)`